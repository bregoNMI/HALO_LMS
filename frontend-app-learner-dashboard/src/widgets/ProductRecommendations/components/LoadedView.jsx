import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Container } from '@openedx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';

import messages from '../messages';
import { courseShape, courseTypeToProductTypeMap } from '../utils';
import ProductCardContainer from './ProductCardContainer';

const LoadedView = ({ crossProductCourses, openCourses }) => {
  const { formatMessage } = useIntl();
  const includesCrossProductTypes = crossProductCourses.length === 2;

  const finalProductList = useMemo(() => {
    if (includesCrossProductTypes) {
      const openCourseList = openCourses.slice(0, 2);
      return crossProductCourses.concat(openCourseList);
    }
    return openCourses;
  }, [crossProductCourses, openCourses, includesCrossProductTypes]);

  const courseTypes = [...new Set(finalProductList.map((item) => courseTypeToProductTypeMap[item.courseType]))];

  return (
    <Container
      size="lg"
      className="recommendations-container pt-sm-5 pt-4.5 pb-2 pb-sm-4.5"
    >
      <h2>
        {formatMessage(messages.recommendationsHeading)}
      </h2>
      <ProductCardContainer finalProductList={finalProductList} courseTypes={courseTypes} />
    </Container>
  );
};

LoadedView.propTypes = {
  crossProductCourses: PropTypes.arrayOf(
    PropTypes.shape(courseShape),
  ).isRequired,
  openCourses: PropTypes.arrayOf(
    PropTypes.shape(courseShape),
  ).isRequired,
};

export default LoadedView;
