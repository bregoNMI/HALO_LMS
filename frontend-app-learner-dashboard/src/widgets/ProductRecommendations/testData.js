export const getCoursesWithType = (courseTypes, parameters = true) => {
  const courses = [];
  const marketingUrl = parameters
    ? 'https://www.edx.org/course/some-course?utm_source=source'
    : 'https://www.edx.org/course/some-course';

  courseTypes.forEach((type) => {
    courses.push({
      title: 'Introduction to Computer Science',
      courseRunKey: 'course-v1:Test+Course+2022T2',
      marketingUrl,
      courseType: type,
      image: {
        src: 'https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg',
      },
      owners: [
        {
          key: 'HarvardX',
          name: 'Harvard University',
          logoImageUrl: 'http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png',
        },
      ],
    });
  });

  return courses;
};

export const mockFooterRecommendationsHook = {
  default: { isExperimentActive: false, inRecommendationsVariant: true },
  activeControl: { isExperimentActive: true, inRecommendationsVariant: false },
  activeTreatment: { isExperimentActive: true, inRecommendationsVariant: true },
};

export const mockCrossProductCourses = getCoursesWithType(['executive-education-2u', 'bootcamp-2u']);
export const mockOpenCourses = getCoursesWithType(['verified-audit', 'audit', 'verified', 'course']);
export const mockFallbackOpenCourse = getCoursesWithType(['course'], false);

export const mockCrossProductResponse = {
  crossProductCourses: mockCrossProductCourses,
  amplitudeCourses: mockOpenCourses,
};

export const mockAmplitudeResponse = {
  amplitudeCourses: mockOpenCourses,
};
