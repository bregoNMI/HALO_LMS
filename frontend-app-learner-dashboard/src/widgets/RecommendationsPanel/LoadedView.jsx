import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Button } from '@openedx/paragon';
import { Search } from '@openedx/paragon/icons';
import { baseAppUrl } from 'data/services/lms/urls';

import { reduxHooks } from 'hooks';
import track from './track';
import CourseCard from './components/CourseCard';
import messages from './messages';

import './index.scss';

export const LoadedView = ({
  courses,
  isControl,
}) => {
  const { formatMessage } = useIntl();
  const { courseSearchUrl } = reduxHooks.usePlatformSettingsData();

  return (
    <div className="p-4 w-100 panel-background">
      <h3 className="pb-2">{isControl === false
        ? formatMessage(messages.recommendationsHeading) : formatMessage(messages.popularCoursesHeading)}
      </h3>
      <div>
        {courses.map((course) => (
          <CourseCard
            key={course.courseKey}
            course={course}
            isControl={isControl}
          />
        ))}
      </div>
      <div className="text-center explore-courses-btn">
        <Button
          variant="tertiary"
          iconBefore={Search}
          as="a"
          href={baseAppUrl(courseSearchUrl)}
          onClick={track.findCoursesWidgetClicked(baseAppUrl(courseSearchUrl))}
        >
          {formatMessage(messages.exploreCoursesButton)}
        </Button>
      </div>
    </div>
  );
};

LoadedView.defaultProps = {
  isControl: true,
};

LoadedView.propTypes = {
  courses: PropTypes.arrayOf(PropTypes.shape({
    courseKey: PropTypes.string,
    title: PropTypes.string,
    logoImageUrl: PropTypes.string,
    marketingUrl: PropTypes.string,
  })).isRequired,
  isControl: PropTypes.oneOf([true, false, null]),
};

export default LoadedView;
