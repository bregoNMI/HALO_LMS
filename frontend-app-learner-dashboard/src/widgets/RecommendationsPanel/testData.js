// eslint-disable-next-line
export default {"courses":[{"courseKey":"ETSx+TOEFLx","logoImageUrl":"https://prod-discovery.edx-cdn.org/organization/logos/9d9e1a30-c34d-4ad1-8c5a-d2410db8c123-8beea336c2a4.png","marketingUrl":"https://www.edx.org/course/toefl-test-preparation-the-insiders-guide?utm_source=lms_catalog_service_user&utm_medium=affiliate_partner","title":"TOEFL® Test Preparation: The Insider’s Guide"},{"courseKey":"HarvardX+CS50P","logoImageUrl":"https://prod-discovery.edx-cdn.org/organization/logos/44022f13-20df-4666-9111-cede3e5dc5b6-2cc39992c67a.png","marketingUrl":"https://www.edx.org/course/cs50s-introduction-to-programming-with-python?utm_source=lms_catalog_service_user&utm_medium=affiliate_partner","title":"CS50's Introduction to Programming with Python"},{"courseKey":"UQx+IELTSx","logoImageUrl":"https://prod-discovery.edx-cdn.org/organization/logos/8554749f-b920-4d7f-8986-af6bb95290aa-f336c6a2ca11.png","marketingUrl":"https://www.edx.org/course/ielts-academic-test-preparation?utm_source=lms_catalog_service_user&utm_medium=affiliate_partner","title":"IELTS Academic Test Preparation"},{"courseKey":"QUx+QU01X02","logoImageUrl":"https://prod-discovery.edx-cdn.org/organization/logos/7b6ca3d5-1030-408f-b1a4-fb140db1854e-7337e1e6deaf.png","marketingUrl":"https://www.edx.org/course/arabic-for-non-arabic-speakers?utm_source=lms_catalog_service_user&utm_medium=affiliate_partner","title":"Arabic for non-Arabic Speakers"},{"courseKey":"HarvardX+CS50W","logoImageUrl":"https://prod-discovery.edx-cdn.org/organization/logos/44022f13-20df-4666-9111-cede3e5dc5b6-2cc39992c67a.png","marketingUrl":"https://www.edx.org/course/cs50s-web-programming-with-python-and-javascript?utm_source=lms_catalog_service_user&utm_medium=affiliate_partner","title":"CS50's Web Programming with Python and JavaScript"}],"isControl":true};
