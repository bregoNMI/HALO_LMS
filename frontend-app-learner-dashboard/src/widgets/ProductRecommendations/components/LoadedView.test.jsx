import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { mockCrossProductCourses, mockOpenCourses } from '../testData';
import LoadedView from './LoadedView';

describe('ProductRecommendations LoadedView', () => {
  it('matches snapshot', () => {
    expect(
      shallow(
        <LoadedView
          crossProductCourses={mockCrossProductCourses}
          openCourses={mockOpenCourses}
        />,
      ).snapshot,
    ).toMatchSnapshot();
  });
  describe('with less than 2 cross product courses', () => {
    it('passes in one course type and 4 open courses to the ProductCardContainer props', () => {
      const wrapper = shallow(
        <LoadedView
          crossProductCourses={[mockCrossProductCourses[0]]}
          openCourses={mockOpenCourses}
        />,
      );

      const productCardContainerProps = wrapper.instance.findByType('ProductCardContainer')[0].props;

      expect(productCardContainerProps.courseTypes.length).toEqual(1);
      expect(productCardContainerProps.courseTypes[0]).toEqual('Course');
      expect(productCardContainerProps.finalProductList).toEqual(mockOpenCourses);
    });
  });
});
