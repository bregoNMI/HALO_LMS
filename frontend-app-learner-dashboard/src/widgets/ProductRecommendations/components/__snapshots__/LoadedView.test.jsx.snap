// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRecommendations LoadedView matches snapshot 1`] = `
<Container
  className="recommendations-container pt-sm-5 pt-4.5 pb-2 pb-sm-4.5"
  size="lg"
>
  <h2>
    You might also like
  </h2>
  <ProductCardContainer
    courseTypes={
      Array [
        "Executive Education",
        "Boot Camp",
        "Course",
      ]
    }
    finalProductList={
      Array [
        Object {
          "courseRunKey": "course-v1:Test+Course+2022T2",
          "courseType": "executive-education-2u",
          "image": Object {
            "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
          },
          "marketingUrl": "https://www.edx.org/course/some-course?utm_source=source",
          "owners": Array [
            Object {
              "key": "HarvardX",
              "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
              "name": "Harvard University",
            },
          ],
          "title": "Introduction to Computer Science",
        },
        Object {
          "courseRunKey": "course-v1:Test+Course+2022T2",
          "courseType": "bootcamp-2u",
          "image": Object {
            "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
          },
          "marketingUrl": "https://www.edx.org/course/some-course?utm_source=source",
          "owners": Array [
            Object {
              "key": "HarvardX",
              "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
              "name": "Harvard University",
            },
          ],
          "title": "Introduction to Computer Science",
        },
        Object {
          "courseRunKey": "course-v1:Test+Course+2022T2",
          "courseType": "verified-audit",
          "image": Object {
            "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
          },
          "marketingUrl": "https://www.edx.org/course/some-course?utm_source=source",
          "owners": Array [
            Object {
              "key": "HarvardX",
              "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
              "name": "Harvard University",
            },
          ],
          "title": "Introduction to Computer Science",
        },
        Object {
          "courseRunKey": "course-v1:Test+Course+2022T2",
          "courseType": "audit",
          "image": Object {
            "src": "https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg",
          },
          "marketingUrl": "https://www.edx.org/course/some-course?utm_source=source",
          "owners": Array [
            Object {
              "key": "HarvardX",
              "logoImageUrl": "http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png",
              "name": "Harvard University",
            },
          ],
          "title": "Introduction to Computer Science",
        },
      ]
    }
  />
</Container>
`;
